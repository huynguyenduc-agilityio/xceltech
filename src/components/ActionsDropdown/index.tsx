import { ReactNode } from 'react';

import { cn } from '@/utils';

import {
  DropdownMenuRoot,
  DropdownMenuTrigger,
  DropdownMenuContent,
  DropdownMenuItem,
} from './base';

export interface ActionsDropdownProps {
  children: React.ReactNode;
  items: {
    key: string;
    label: ReactNode;
    onClick?: () => void;
    disabled?: boolean;
  }[];
  className?: string;
  itemClassName?: string;
  disabled?: boolean;
}

const ActionsDropdown = ({
  children,
  items,
  className,
  itemClassName,
  disabled = false,
}: ActionsDropdownProps) => {
  return (
    <DropdownMenuRoot>
      <DropdownMenuTrigger asChild disabled={disabled}>
        {children}
      </DropdownMenuTrigger>
      <DropdownMenuContent className={cn(className)}>
        {items.map((item) => (
          <DropdownMenuItem
            key={item.key}
            onClick={item.onClick}
            disabled={item.disabled}
            className={itemClassName}
          >
            {item.label}
          </DropdownMenuItem>
        ))}
      </DropdownMenuContent>
    </DropdownMenuRoot>
  );
};

export default ActionsDropdown;
