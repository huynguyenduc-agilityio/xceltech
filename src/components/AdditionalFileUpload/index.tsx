import { useCallback, useState } from 'react';
import { useForm } from 'react-hook-form';

import {
  Button,
  FormField,
  FormItem,
  FormMessage,
  Input,
  Label,
} from '../common';

export interface AdditionalFileUploadProps {
  name: string;
  fileUrl?: string;
  onFileChange: (file: string) => void;
}

const AdditionalFileUpload = ({
  name,
  fileUrl = '',
}: AdditionalFileUploadProps) => {
  const [selectedFile] = useState<string | null>(fileUrl);

  const {
    control,
    setValue,
    formState: { errors },
  } = useForm<Record<string, File>>({
    mode: 'onBlur',
    reValidateMode: 'onChange',
  });

  const handleFileChange = useCallback(
    async (e: React.ChangeEvent<HTMLInputElement>) => {
      const file = e.target.files?.[0];

      if (file) {
        setValue(name, file, { shouldValidate: true });

        //TODO: Hanlde Upload File
      }
    },
    [setValue],
  );

  return (
    <div className="flex-1 w-full">
      <FormField
        control={control}
        name={name}
        render={() => (
          <FormItem>
            <div className="relative flex items-center h-[68px]">
              <Input
                id="file-upload"
                type="file"
                className="hidden"
                onChange={handleFileChange}
                accept=".pdf"
              />
              <div className="absolute inset-0 flex items-center px-6 w-full h-full bg-blue-light rounded-[15px] test-huy">
                <span className="mr-[291px] text-md truncate text-black-smoky-">
                  {selectedFile}
                </span>
              </div>
              <Button
                variant="secondary"
                className="absolute bottom-0 right-0 w-[291px] py-[12px] h-full rounded-[15px]"
              >
                <Label
                  htmlFor="file-upload"
                  className="text-lg text-bold cursor-pointer"
                >
                  Upload
                </Label>
              </Button>
            </div>
            {errors?.file && (
              <FormMessage className="">
                <div className="flex flex-col gap-4 text-red-500 text-sm">
                  <p>{errors.file.message}</p>
                  <p>Please select again!!!</p>
                </div>
              </FormMessage>
            )}
          </FormItem>
        )}
      />
    </div>
  );
};

export default AdditionalFileUpload;
