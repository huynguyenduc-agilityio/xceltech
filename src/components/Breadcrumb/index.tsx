import React from 'react';

type BreadcrumbItem = {
  label: string;
  href?: string;
};

type BreadcrumbProps = {
  items: BreadcrumbItem[];
};

const Breadcrumb = ({ items }: BreadcrumbProps) => {
  return (
    <nav className="flex items-center gap-2 text-xl text-black-soft">
      {items.map((item, index) => (
        <React.Fragment key={index}>
          {/* TODO: Convert <a> to <Link> */}
          {item.href ? (
            <a href={item.href} className="hover:underline">
              {item.label}
            </a>
          ) : (
            <span>{item.label}</span>
          )}
          {index < items.length - 1 && '>'}
        </React.Fragment>
      ))}
    </nav>
  );
};

export default Breadcrumb;
