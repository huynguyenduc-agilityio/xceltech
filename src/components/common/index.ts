export * from './Input';
export * from './Label';
export * from './Form';
export * from './Button';
export * from './Calendar';
export * from './Popover';
export * from './Avatar';
export * from './Textarea';
export * from './Progress';
export * from './Select';
export * from './Checkbox';
export * from './Carousel';
export * from './Dialog';
export { default as DatePicker } from './DatePicker';
export { default as Toast } from './Toast';
