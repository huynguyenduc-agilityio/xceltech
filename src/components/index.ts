export * from './common';
export { default as AdditionalFileUpload } from './AdditionalFileUpload';
export { default as Breadcrumb } from './Breadcrumb';
export { default as DocumentUpload } from './DocumentUpload';
export { default as InfoTile } from './InfoTile';
export { default as LeaveTile } from './LeaveTile';
export { default as Navbar } from './Navbar';
export { default as OverviewCard } from './OverviewCard';
export { default as SideMenu } from './SideMenu';
export { default as Sidebar } from './Sidebar';
export { default as TextField } from './TextField';
export { default as NotificationBadge } from './NotificationBadge';
export { default as Status } from './Status';
export { default as ImageUpload } from './ImageUpload';
export { default as ConfirmDialog } from './ConfirmDialog';
export { default as Table } from './Table';
export { default as SearchBar } from './SearchBar';
export { default as Fallback } from './Fallback';
