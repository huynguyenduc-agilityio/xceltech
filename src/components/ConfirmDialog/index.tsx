import { useConfirmDialogStore } from '@/stores';

import { Button, DialogContainer } from '../common';

const ConfirmDialog = () => {
  const { isOpen, title, confirmMessage, onConfirm, onCancel, close } =
    useConfirmDialogStore();

  if (!isOpen) return null;

  const handleConfirm = () => {
    onConfirm?.();
    close();
  };

  const handleCancel = () => {
    onCancel?.();
    close();
  };

  return (
    <DialogContainer
      isOpen={isOpen}
      title={title}
      description={confirmMessage}
      footer={
        <>
          <Button
            variant="outline"
            className="text-sm h-fit px-4 py-2"
            onClick={handleCancel}
          >
            Cancel
          </Button>
          <Button className="text-sm h-fit px-4 py-2" onClick={handleConfirm}>
            Confirm
          </Button>
        </>
      }
    />
  );
};

export default ConfirmDialog;
