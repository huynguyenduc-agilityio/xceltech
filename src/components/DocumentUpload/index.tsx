import { useCallback, useEffect, useState } from 'react';

import { Button, Input, Label } from '../common';

export interface DocumentUploadProps {
  value?: File | string | null;
  onFileChange: (file: File) => void;
}

const DocumentUpload = ({ value, onFileChange }: DocumentUploadProps) => {
  const [fileUrl, setFileUrl] = useState<string | null>(null);

  const handleFileChange = useCallback(
    async (e: React.ChangeEvent<HTMLInputElement>) => {
      const file = e.target.files?.[0];

      if (file) {
        onFileChange(file);
      }
    },
    [onFileChange],
  );

  useEffect(() => {
    if (value instanceof File) {
      const url = URL.createObjectURL(value);

      setFileUrl(url);

      return () => {
        URL.revokeObjectURL(url);
      };
    }

    if (typeof value === 'string') {
      setFileUrl(value);
    }

    if (!value) {
      setFileUrl(null);
    }
  }, [value]);

  const fileName =
    typeof value === 'string'
      ? (value.split('/').pop() ?? '')
      : value instanceof File
        ? value.name
        : '';

  return (
    <div className="relative flex items-center h-[54px]">
      <Button
        asChild
        className="absolute inset z-10 bg-black-shadow hover:bg-black-shadow w-[254px] py-[12px] text-white h-full rounded"
      >
        <Label htmlFor="file-upload" className="cursor-pointer">
          Choose File
        </Label>
      </Button>
      <Input
        id="file-upload"
        type="file"
        className="hidden"
        onChange={handleFileChange}
        accept=".pdf,.jpg,.png,.docx"
      />
      <div className="absolute flex items-center px-6 w-full h-full bg-blue-light rounded-regular">
        <span className="ml-[254px] text-md truncate text-black-smoky-">
          {fileUrl ? (
            <a href={fileUrl} target="_blank" rel="noopener noreferrer">
              {fileName}
            </a>
          ) : (
            ''
          )}
        </span>
      </div>
    </div>
  );
};

export default DocumentUpload;
