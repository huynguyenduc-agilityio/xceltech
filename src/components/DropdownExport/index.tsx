import { useCallback } from 'react';

// Icons
import { CircleDownIcon } from '@/icons';

import { Button } from '../common';
import ActionsDropdown from '../ActionsDropdown';

// Hooks
import { useExportLeaveFile, useToast } from '@/hooks';

// Types
import { ErrorType, ToastStatus } from '@/types';

// Constants
import { FileType, MESSAGES } from '@/constants';

const DropdownExport = () => {
  const { handleExportLeaveFile, isLoading } = useExportLeaveFile();
  const { toast } = useToast();

  const handleClickExportFile = useCallback(
    async (type: FileType) => {
      try {
        await handleExportLeaveFile({ type });

        toast({
          status: ToastStatus.Success,
          title: MESSAGES.COMMON.EXPORT_SUCCESS,
        });
      } catch (error) {
        toast({
          status: ToastStatus.Error,
          title: (error as ErrorType).detail || MESSAGES.COMMON.EXPORT_FAILED,
        });
      }
    },
    [handleExportLeaveFile, toast],
  );

  return (
    <ActionsDropdown
      items={[
        {
          key: 'pdf',
          label: 'PDF',
          onClick: () => handleClickExportFile(FileType.Pdf),
        },
        {
          key: 'csv',
          label: 'CSV',
          onClick: () => handleClickExportFile(FileType.Csv),
        },
        {
          key: 'excel',
          label: 'Excel',
          onClick: () => handleClickExportFile(FileType.Excel),
        },
      ]}
      itemClassName="bg-white text-black-default border-b border-b-gray-200 hover:bg-gray-100"
    >
      <Button
        className="w-[195px] h-14 bg-green-primary hover:bg-green-primary/90 shadow-md"
        isLoading={isLoading}
      >
        Export
        <CircleDownIcon className="ml-4" />
      </Button>
    </ActionsDropdown>
  );
};

export default DropdownExport;
