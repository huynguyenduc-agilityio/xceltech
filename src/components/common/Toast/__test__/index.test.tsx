import { render } from '@testing-library/react';
import { Toast } from '@/components';
import { useToast } from '@/hooks';
import { ToastStatus } from '@/types';

jest.mock('@/hooks', () => ({
  useToast: jest.fn(),
}));

describe('Toast Component', () => {
  it('renders success toast correctly', () => {
    (useToast as jest.Mock).mockReturnValue({
      toasts: [
        {
          id: '1',
          status: ToastStatus.Success,
          title: 'Success!',
          description: 'Your action was successful.',
        },
      ],
    });

    const { asFragment } = render(<Toast />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders error toast correctly', () => {
    (useToast as jest.Mock).mockReturnValue({
      toasts: [
        {
          id: '2',
          status: ToastStatus.Error,
          title: 'Error!',
          description: 'Something went wrong.',
        },
      ],
    });

    const { asFragment } = render(<Toast />);
    expect(asFragment()).toMatchSnapshot();
  });
});
