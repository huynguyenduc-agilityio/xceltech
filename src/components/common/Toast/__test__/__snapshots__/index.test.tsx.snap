// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast Component renders error toast correctly 1`] = `
<DocumentFragment>
  <div
    aria-label="Notifications (F8)"
    role="region"
    style=""
    tabindex="-1"
  >
    <span
      aria-hidden="true"
      style="position: fixed; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
      tabindex="0"
    />
    <ol
      class="fixed top-0 right-0 gap-2 z-[100] flex max-h-screen w-full flex-col p-4 sm:right-0 md:max-w-[420px] overflow-hidden"
      tabindex="-1"
    >
      <li
        aria-atomic="true"
        aria-live="off"
        class="group pointer-events-auto relative flex w-full items-center justify-between space-x-4 overflow-hidden transition-all data-[swipe=cancel]:translate-x-0 data-[swipe=end]:translate-x-[var(--radix-toast-swipe-end-x)] data-[swipe=move]:translate-x-[var(--radix-toast-swipe-move-x)] data-[swipe=move]:transition-none data-[state=open]:animate-in data-[state=closed]:animate-out data-[swipe=end]:animate-out data-[state=closed]:fade-out-80 data-[state=closed]:slide-out-to-right-full data-[state=open]:slide-in-from-top-full data-[state=open]:sm:slide-in-from-bottom-full border text-foreground border-l-8 p-4 shadow-lg rounded-lg border-red-600 bg-red-100"
        data-radix-collection-item=""
        data-state="open"
        data-swipe-direction="right"
        role="status"
        style="user-select: none;"
        tabindex="0"
      >
        <div
          class="flex items-center gap-3"
        >
          <div
            class="flex item-center w-6"
          >
            <div
              class="flex justify-center items-center w-6 h-6 rounded-full bg-red-600"
            >
              <svg
                class="lucide lucide-x text-white w-4 h-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 6 6 18"
                />
                <path
                  d="m6 6 12 12"
                />
              </svg>
            </div>
          </div>
          <div
            class="grid gap-1"
          >
            <div
              class="text-sm font-semibold"
            >
              Error!
            </div>
            <div
              class="text-sm opacity-90"
            >
              Something went wrong.
            </div>
          </div>
        </div>
        <button
          class="absolute right-2 top-2 rounded-md p-1 text-foreground/50 opacity-0 transition-opacity hover:text-foreground focus:opacity-100 focus:outline-none focus:ring-2 group-hover:opacity-100 group-[.destructive]:text-red-300 group-[.destructive]:hover:text-red-50 group-[.destructive]:focus:ring-red-400 group-[.destructive]:focus:ring-offset-red-600"
          data-radix-toast-announce-exclude=""
          toast-close=""
          type="button"
        >
          <svg
            class="lucide lucide-x h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 6 6 18"
            />
            <path
              d="m6 6 12 12"
            />
          </svg>
        </button>
      </li>
    </ol>
    <span
      aria-hidden="true"
      style="position: fixed; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
      tabindex="0"
    />
  </div>
</DocumentFragment>
`;

exports[`Toast Component renders success toast correctly 1`] = `
<DocumentFragment>
  <div
    aria-label="Notifications (F8)"
    role="region"
    style=""
    tabindex="-1"
  >
    <span
      aria-hidden="true"
      style="position: fixed; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
      tabindex="0"
    />
    <ol
      class="fixed top-0 right-0 gap-2 z-[100] flex max-h-screen w-full flex-col p-4 sm:right-0 md:max-w-[420px] overflow-hidden"
      tabindex="-1"
    >
      <li
        aria-atomic="true"
        aria-live="off"
        class="group pointer-events-auto relative flex w-full items-center justify-between space-x-4 overflow-hidden transition-all data-[swipe=cancel]:translate-x-0 data-[swipe=end]:translate-x-[var(--radix-toast-swipe-end-x)] data-[swipe=move]:translate-x-[var(--radix-toast-swipe-move-x)] data-[swipe=move]:transition-none data-[state=open]:animate-in data-[state=closed]:animate-out data-[swipe=end]:animate-out data-[state=closed]:fade-out-80 data-[state=closed]:slide-out-to-right-full data-[state=open]:slide-in-from-top-full data-[state=open]:sm:slide-in-from-bottom-full border text-foreground border-l-8 p-4 shadow-lg rounded-lg border-green-primary bg-green-100"
        data-radix-collection-item=""
        data-state="open"
        data-swipe-direction="right"
        role="status"
        style="user-select: none;"
        tabindex="0"
      >
        <div
          class="flex items-center gap-3"
        >
          <div
            class="flex item-center w-6"
          >
            <div
              class="flex justify-center items-center w-6 h-6 rounded-full bg-green-primary"
            >
              <svg
                class="lucide lucide-check text-white w-4 h-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M20 6 9 17l-5-5"
                />
              </svg>
            </div>
          </div>
          <div
            class="grid gap-1"
          >
            <div
              class="text-sm font-semibold"
            >
              Success!
            </div>
            <div
              class="text-sm opacity-90"
            >
              Your action was successful.
            </div>
          </div>
        </div>
        <button
          class="absolute right-2 top-2 rounded-md p-1 text-foreground/50 opacity-0 transition-opacity hover:text-foreground focus:opacity-100 focus:outline-none focus:ring-2 group-hover:opacity-100 group-[.destructive]:text-red-300 group-[.destructive]:hover:text-red-50 group-[.destructive]:focus:ring-red-400 group-[.destructive]:focus:ring-offset-red-600"
          data-radix-toast-announce-exclude=""
          toast-close=""
          type="button"
        >
          <svg
            class="lucide lucide-x h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 6 6 18"
            />
            <path
              d="m6 6 12 12"
            />
          </svg>
        </button>
      </li>
    </ol>
    <span
      aria-hidden="true"
      style="position: fixed; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
      tabindex="0"
    />
  </div>
</DocumentFragment>
`;
